retry.ml

```diff
@@ -0,0 +1,101 @@
+open Prelude
+
+let log = Log.from "retry"
+
+(** Clamped random exponential backoff for retry *)
+let exp_backoff_pause ?max_delay attempt =
```

Comment:
max_delay should be required here? @jorisgio

Reply:
it seems to always be used with this parameter specified in our codebase, I'm making it required.